components/tabs/tabs.js

```diff
@@ -0,0 +1,840 @@
+		} else {
+			// make sure there will not be any empty space between right side of container and first tab
+			if (newTranslationValue < 0) newTranslationValue = 0;
+		}
+
+		// make sure the new position will not place selected tab behind the right scroll button
+		if (this.dir !== 'rtl') {
+			expectedPosition = selectedTabMeasures.offsetLeft + selectedTabMeasures.rect.width + newTranslationValue;
+			if ((selectedTabIndex < this._tabInfos.length - 1) && this._isPositionInRightScrollArea(expectedPosition, measures)) {
+				newTranslationValue = getNewTranslationValue();
+			}
+		} else {
+			expectedPosition = selectedTabMeasures.offsetLeft + newTranslationValue;
+			if ((selectedTabIndex < this._tabInfos.length - 1) && this._isPositionInLeftScrollArea(expectedPosition)) {
+				newTranslationValue = getNewTranslationValue();
+			}
+		}
+
+		return newTranslationValue;
+
+	}
+
+	async _focusSelected() {
+		const selectedTab = this.shadowRoot.querySelector('d2l-tab-internal[aria-selected="true"]');
+		if (!selectedTab) return;
+
+		const selectedTabInfo = this._getTabInfo(selectedTab.controlsPanel);
+		await this._updateScrollPosition(selectedTabInfo);
+
+		selectedTab.focus();
+	}
+
+	_getComputedBackgroundColor() {
+		let bgColor = null;
+
+		findComposedAncestor(this, (node) => {
+			if (!node || node.nodeType !== Node.ELEMENT_NODE) return false;
+			const nodeColor = getComputedStyle(node, null)['backgroundColor'];
+			if (nodeColor === 'rgba(0, 0, 0, 0)' || nodeColor === 'transparent') return false;
+			bgColor = nodeColor;
+			return true;
+		});
+
+		return bgColor;
+	}
+
+	_getMeasures() {
+		if (!this._measures) this._updateMeasures();
+		return this._measures;
+	}
+
+	_getPanel(id) {
+		// use simple selector for slot (Edge)
+		const slot = this.shadowRoot.querySelector('.d2l-panels-container').querySelector('slot');
+		const panels = this._getPanels(slot);
+		for (let i = 0; i < panels.length; i++) {
+			if (panels[i].nodeType === Node.ELEMENT_NODE && panels[i].role === 'tabpanel' && panels[i].id === id) {
+				return panels[i];
+			}
+		}
+	}
+
+	_getPanels(slot) {
+		if (!slot) return;
+		return slot.assignedNodes({flatten: true})
+			.filter((node) => node.nodeType === Node.ELEMENT_NODE && node.role === 'tabpanel');
+	}
+
+	_getTabInfo(id) {
+		return this._tabInfos.find((t) => t.id === id);
+	}
+
+	_handleFocusEnd(e) {
+		if (e.relatedTarget && e.relatedTarget.role !== 'tab') {
+			this._focusSelected();
+		} else {
+			const nextFocusable = getNextFocusable(e.target, false);
+			if (nextFocusable) nextFocusable.focus();
+		}
+	}
+
+	_handleFocusStart(e) {
+		if (e.relatedTarget && e.relatedTarget.role === 'tab') {
+			const previousFocusable = getPreviousFocusable(e.target, false);
+			if (previousFocusable) previousFocusable.focus();
+		} else {
+			this._focusSelected();
+		}
+	}
+
+	async _handlePanelsSlotChange(e) {
+
+		const panels = this._getPanels(e.target);
+		/*
+		// do fancy remove the panels
```

Comment:
Was this meant to be left commented out?

Reply:
Just about to push some changes that includes this.  :)